### ISSUE TITLE
Don't suggest use mutable keyword for parameters

### ISSUE THREAD

**Issue by psfinaki** [dotnet/fsharp#15803]
This is misleading:
![image](https://github.com/dotnet/fsharp/assets/5451366/56ffa713-154c-4cc8-ab71-58c00f16579a)

The error is okay I think, we just shouldn't suggest making parameters mutable, this is illegal.

**Comment by psfinaki** [comment]
@Happypig375 mhm this would be compilable code indeed, I just wonder if there can really be a valid use case for that - as in, any real world code where you would _recommend_ a developer such an approach instead of e.g. refactoring code to remove mutables :)

**Comment by smoothdeveloper** [comment]
@psfinaki, it would make sense to have another error message for this case:

Suggestion, but there is probably better way to phrase it.

> The `x` binding is an immutable parameter, consider defining a local mutable binding or adjusting the parameter to byref.

related: #1103

**Comment by psfinaki** [comment]
Yeah I think I would be cautious with the word "consider" and be more like "if you for some weird reason intended to ... then ..." :) But yeah, that's details, more important it should be just a different diag/message.

**Comment by vzarytovskii** [comment]
> @Happypig375 mhm this would be compilable code indeed, I just wonder if there can really be a valid use case for that - as in, any real world code where you would _recommend_ a developer such an approach instead of e.g. refactoring code to remove mutables :)

Yes, when you want to make defensive copy.

**Comment by voronoipotato** [comment]
Suggesting making the parameter mutable is more hostile to beginners than it is helpful to experienced F# devs. We can acknowledge that it's possible without actively jumping to a recommendation. 

```
The x binding is an immutable parameter, consider defining a new variable or adjusting the parameter to byref.
```

This way, it doesn't actively encourage a mutable copy when for most beginners learning the language probably would have an easier time with a new immutable variable.
